### DISCUSSION TITLE
upgrade python version in dockerimage

### DISCUSSION THREAD

**Discussion by tooptoop4** [apache/superset#22810] [Q&A / Help]
can we change to 3.8  without minor version to get newer base?
https://github.com/apache/superset/blob/master/Dockerfile#L21

**Comment by rusackas** [comment]
I assume you mean patch version rather than minor version? Either way, I'd ask @villebro who bumped it to 3.8.12, @nisheldo who bumped it to 3.8.13, and @EugeneTorap who updated it to the `slim` version. Any reason we can't un-pin it?

**Reply by villebro** [reply]
I kind of agree that pinning to [3.8-slim](https://hub.docker.com/layers/library/python/3.8-slim/images/sha256-7c11ba98b104bed6dec12a3b10cb59b8dd1fa47ae1bae1ced6b609d5cacca8fc?context=explore) (or rather move on to [3.9-slim](https://hub.docker.com/layers/library/python/3.9-slim/images/sha256-8b0502fe3a8ae9d26567f66e217223595ff6abe901cba64fd8b1b58176311b27?context=explore) while at it) could make sense. Thoughts?

**Reply by EugeneTorap** [reply]
It's better to migrate from 3.8 to 3.9 and set [3.9-slim](https://hub.docker.com/layers/library/python/3.9-slim/images/sha256-8b0502fe3a8ae9d26567f66e217223595ff6abe901cba64fd8b1b58176311b27?context=explore) in Dockerfile

**Reply by villebro** [reply]
Here's my proposal: After we cut 2.1, we
- drop official support for 3.8 and mention this in the 2.1 release notes
- migrate CI from 3.8 and 3.9 to 3.9 and 3.10
- Replace 3.8.13-slim with 3.9-slim

**Reply by rusackas** [reply]
Just to say it out loud, this could turn into a 2.1 thing and/or a 3.0 thing, depending on how the timing plays out.

**Reply by tooptoop4** [reply]
can at least 3.8-slim get into 2.0.2 ? that would get the CVEs cleared

**Reply by EugeneTorap** [reply]
@villebro I think it will be a breaking change if we drop official support for 3.8 python because users who use python 3.8 in superset 2.0.1 will have problem with updating to minor version of superset.
@tooptoop4 @villebro I have prepared a #22934 PR in order to pin the python image to to 3.8-slim in Dockerfile. Pls, review it.
